pandas/core/frame.py

```diff
@@ -835,15 +835,129 @@ def __len__(self):
 
     def dot(self, other):
         """
-        Matrix multiplication with DataFrame or Series objects
+        Find the dot product of two matrices.
+
+        Using two DataFrame, matrix, or Series objects -
+        this method finds the standard inner product (dot product)
+        of two conformable matrices. Matrices A and B are said
+        to be conformable for multiplication A(B) whenever the
+        number of columns in A = the number of rows in B.
+        (i.e. -  A is m × p and B is p × n)
 
         Parameters
         ----------
-        other : DataFrame or Series
+        other : DataFrame, matrix, or Series
+            Different combinations shown in examples below.
 
         Returns
         -------
-        dot_product : DataFrame or Series
+        dot_product : DataFrame, matrix, or Series
+            Return type depends on input, see examples below.
+
+        Raises
+        ------
+        ValueError
+            If the first dimension of A is not the same size as the
+            secondi dimension of B.
+
+        See Also
+        --------
+        numpy.dot : dot product in numpy
+
+        Examples
+        --------
+        Two DataFrame objects:
+
+        >>> df_A = pd.DataFrame([[1,2],[3,4]])
+        >>> df_A
+           0  1
+        0  1  2
+        1  3  4
+
+        >>> df_B = pd.DataFrame([[3,4],[1,2]])
+        >>> df_B
+           0  1
+        0  3  4
+        1  1  2
+
+        >>> df_A.dot(df_B)
+              0   1
+        0     5   8
+        1    13   20
+
+        A DataFrame and a matrix object (returns DataFrame):
+
+        >>> df_A = pd.DataFrame([[1,2],[3,4]])
+        >>> df_A
+           0  1
+        0  1  2
+        1  3  4
+
+        >>> mat_B = np.matrix('3 4 ; 1 2')
+        >>> mat_B
+        matrix([[3, 4],
+               [1, 2]])
+
+        >>> df_A.dot(mat_B)
+              0   1
+        0     5   8
+        1    13   20
```

Comment:
I really like that you already mentioned the resulting data type in the header. Maybe add a demonstration to the end of very first example (so a new user know how to check that)?
`type(df_A.dot(df_B))`